program-runtime/src/compute_budget.rs

```diff
@@ -261,18 +261,20 @@ impl ComputeBudget {
             self.heap_size = Some(bytes as usize);
         }
 
-        self.compute_unit_limit = if default_units_per_instruction {
-            updated_compute_unit_limit.or_else(|| {
-                Some(
-                    (num_non_compute_budget_instructions as u32)
-                        .saturating_mul(DEFAULT_INSTRUCTION_COMPUTE_UNIT_LIMIT),
-                )
-            })
-        } else {
-            updated_compute_unit_limit
-        }
-        .unwrap_or(MAX_COMPUTE_UNIT_LIMIT)
-        .min(MAX_COMPUTE_UNIT_LIMIT) as u64;
+        self.compute_unit_limit = u64::from(
+            if default_units_per_instruction {
+                updated_compute_unit_limit.or_else(|| {
+                    Some(
+                        num_non_compute_budget_instructions
+                            .saturating_mul(DEFAULT_INSTRUCTION_COMPUTE_UNIT_LIMIT),
+                    )
+                })
+            } else {
+                updated_compute_unit_limit
+            }
+            .unwrap_or(MAX_COMPUTE_UNIT_LIMIT)
+            .min(MAX_COMPUTE_UNIT_LIMIT),
+        );
```

Comment:
```suggestion
        let compute_unit_limit = if default_units_per_instruction {
            updated_compute_unit_limit.or_else(|| {
                Some(
                    num_non_compute_budget_instructions
                        .saturating_mul(DEFAULT_INSTRUCTION_COMPUTE_UNIT_LIMIT),
                )
            })
        } else {
            updated_compute_unit_limit
        }
        .unwrap_or(MAX_COMPUTE_UNIT_LIMIT)
        .min(MAX_COMPUTE_UNIT_LIMIT);
        self.compute_unit_limit = u64::from(compute_unit_limit);
```

easier to read, shorter diff